python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,346 @@
+##########################################################################
+# Ganga Project. http://cern.ch/ganga
+#
+# $Id: Executable.py,v 1.1 2008-07-17 16:40:57 moscicki Exp $
+##########################################################################
+
+from Ganga.GPIDev.Adapters.IPrepareApp import IPrepareApp
+from Ganga.GPIDev.Adapters.IRuntimeHandler import IRuntimeHandler
+from Ganga.GPIDev.Schema import Schema, Version, SimpleItem
+
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.File import File, ShareDir
+from Ganga.GPIDev.Lib.File.FileBuffer import FileBuffer
+from Ganga.Core import ApplicationConfigurationError, ApplicationPrepareError
+
+from Ganga.Utility.logging import getLogger
+
+from Ganga.GPIDev.Base.Proxy import getName
+
+import os
+import shutil
+from datetime import datetime
+import tempfile
+import tarfile
+import random
+import threading
+from Ganga.Utility.files import expandfilename
+
+from Ganga.GPIDev.Adapters.ApplicationRuntimeHandlers import allHandlers
+
+from GangaGaudi.Lib.RTHandlers.RunTimeHandlerUtils import get_share_path, master_sandbox_prepare, sandbox_prepare, script_generator
+from GangaDirac.Lib.RTHandlers.DiracRTHUtils import dirac_inputdata, dirac_ouputdata, mangle_job_name, diracAPI_script_template, diracAPI_script_settings, API_nullifier, dirac_outputfile_jdl
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+from Ganga.GPIDev.Lib.File.OutputFileManager import getOutputSandboxPatterns, getWNCodeForOutputPostprocessing
+from Ganga.GPIDev.Adapters.StandardJobConfig import StandardJobConfig
+from Ganga.Utility.util import unique
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+
+logger = getLogger()
+
+data_file = 'data.py'
+
+prep_lock = threading.Lock()
+
+def add_timeStampFile(given_path):
+    """
+    This creates a file in this directory given called __timestamp__ which contains the time so that the final file is unique
+    Args:
+        given_path (str): Path which we want to create the timestamp within
+
+    """
+    fmt = '%Y-%m-%d-%H-%M-%S'
+    time_filename = os.path.join(given_path, '__timestamp__')
+    logger.info("Constructing: %s" % time_filename)
+    with open(time_filename, 'a+') as time_file:
+        time_file.write(datetime.now().strftime(fmt))
+
+def genDataFiles(job):
+    """
+    Generating a data.py file which contains the data we want gaudirun to use
+    Args:
+        job (Job): This is the job object which contains everything useful for generating the code
+    """
+    logger.debug("Doing XML Catalog stuff")
+
+    inputsandbox = []
+
+    data = job.inputdata
+    data_str = ''
```

Comment:
Given this is no longer used outside of `if data:` I'll just remove this for now